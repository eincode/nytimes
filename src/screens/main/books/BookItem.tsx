import React, { StatelessComponent } from "react"
import {
  TouchableOpacity,
  View,
  Text,
  StyleSheet,
  TouchableOpacityProps,
  Image,
} from "react-native"
import metrics from "../../../config/metrics"

interface IProps extends TouchableOpacityProps {
  book?: BookResult
}

const BookItem: StatelessComponent<IProps> = (props: IProps) => (
  <TouchableOpacity {...props} style={styles.container}>
    <Image
      source={{
        uri: `https://s1.nyt.com/du/books/images/${
          props.book!.isbns[0]
            ? props.book!.isbns[0].isbn13
            : props.book!.book_details[0].primary_isbn13
        }.jpg`,
      }}
      style={styles.image}
      resizeMode={"cover"}
    />
    <View style={styles.contentContainer}>
      <Text style={{ fontSize: 15, fontWeight: "bold" }}>
        {props.book!.book_details[0].title}
      </Text>
      <Text style={{ color: "grey", fontSize: 12 }}>
        by {props.book!.book_details[0].author}
      </Text>
      <Text style={{ marginTop: 10 }}>
        {props.book!.book_details[0].description}
      </Text>
    </View>
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  container: {
    maxWidth: metrics.DEVICE_WIDTH * 0.45,
    backgroundColor: "white",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowRadius: 5,
    shadowOpacity: 1,
    elevation: 5,
    borderRadius: 5,
    padding: 10,
    minHeight: 200,
    flex: 1,
    margin: 10,
  },

  image: {
    width: metrics.DEVICE_WIDTH * 0.45,
    height: 200,
    position: "absolute",
  },

  contentContainer: {
    marginTop: 200,
  },
})

BookItem.defaultProps = {
  book: {
    amazon_product_url:
      "https://www.amazon.com/Full-Package-Lauren-Blakely-ebook/dp/B01MT5HMRV?tag=NYTBS-20",
    isbns: [],
    book_details: [
      {
        author: "book_author",
        description: "book_description",
        primary_isbn13: "9780385341004",
        publisher: "book_publisher",
        title: "book_title",
      },
    ],
  },
}

export default BookItem
