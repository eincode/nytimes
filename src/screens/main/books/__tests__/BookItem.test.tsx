import React from "react"
import renderer from "react-test-renderer"

import BookItem from "../BookItem"

it("Book item component renders correctly", () => {
  const book = renderer.create(<BookItem />).toJSON()
  expect(book).toMatchSnapshot()
})
