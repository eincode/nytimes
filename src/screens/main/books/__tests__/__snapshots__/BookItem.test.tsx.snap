// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book item component renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 5,
      "elevation": 5,
      "flex": 1,
      "margin": 10,
      "maxWidth": 337.5,
      "minHeight": 200,
      "opacity": 1,
      "padding": 10,
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 5,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "https://s1.nyt.com/du/books/images/9780385341004.jpg",
      }
    }
    style={
      Object {
        "height": 200,
        "position": "absolute",
        "width": 337.5,
      }
    }
  />
  <View
    style={
      Object {
        "marginTop": 200,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      book_title
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 12,
        }
      }
    >
      by 
      book_author
    </Text>
    <Text
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      book_description
    </Text>
  </View>
</View>
`;
